website_field_autocomplete/static/src/js/field_autocomplete.js

```diff
@@ -10,44 +10,39 @@ odoo.define('website_field_autocomplete.field_autocomplete', function(require){
 
   snippet_animation.registry.field_autocomplete = snippet_animation.Class.extend({
 
-    selector: '.s_website_autocomplete',
-
+    selector: '.js_website_autocomplete',
+
+    /* Query remote server for autocomplete suggestions
+     * @param request object Request from jQueryUI Autocomplete
+     * @param response function Callback for response, accepts array of str
+     */
+    autocomplete: function(self, request, response) {
+      var QueryModel = new Model(self.$target.data('model'));
+      var queryField = self.$target.data('query-field') || 'name';
+      var displayField = self.$target.data('display-field') || queryField;
+      var limit = self.$target.data('limit') || 10;
+      var domain = [[queryField, 'ilike', request.term]];
```

Comment:
To make it even better, you could move most of these declarations to `start()` and save them as variables under `this`. Submodules would not need to redeclare them.